### PR TITLE
Update and simplify getDirtyIDs method

### PR DESCRIPTION
**Pull request by michalkleiner**
This is a PR updating the code as suggested by @tractorcow in the original issue.

https://github.com/silverstripe/silverstripe-fulltextsearch/issues/108

### CODE REVIEW

**Comment by robbieaverill** [comment]
The code looks much tidier! The unit test failure looks relevant though, would you mind investigating that?

File: src/Search/Indexes/SearchIndex.php [hunk 1]
```diff
@@ -615,23 +615,15 @@ public function getDirtyIDs($class, $id, $statefulids, $fields)
...
-                        // Use TableName for queries
-                        $otherTableName = DataObject::getSchema()->tableName($step['otherclass']);
-
-                        $sql = new SQLSelect('"' . $tableName . '"."ID"', '"' . $tableName . '"', '"' . $otherTableName . '"."ID" IN (' . implode(',', $ids) . ')');
-                        $sql->addInnerJoin($otherTableName, '"' . $tableName . '"."ID" = "' . $otherTableName . '"."' . $step['foreignkey'] . '"');
-                        singleton($step['class'])->extend('augmentSQL', $sql);
-
-                        $ids = $sql->execute()->column();
+                        // Get many ids by querying component with alternate set of foreign ids
+                        $ids = DataObject::singleton($step['otherclass'])
```
[the remark this answers is not in the document]
Comment (NightJar) [reply src/Search/Indexes/SearchIndex.php]: Thanks @UndefinedOffset 
Feel like updating @michalkleiner ?

[the remark this answers is not in the document]
Comment (michalkleiner) [reply src/Search/Indexes/SearchIndex.php]: on it

**Comment by robbieaverill** [comment]
They didn't unfortunately

**Comment by robbieaverill** [comment]
@michalkleiner any thoughts on the test failures?

**Comment by michalkleiner** [comment]
Will try to look at it this week, been away for over 2 weeks.

**Comment by michalkleiner** [comment]
Another month passed and.... here we are again.

**Comment by emteknetnz** [comment]
Hi, we have closed this pull request as there has not been any activity on it for a long time.

We are doing this to make it clearer for everyone in the community which pull requests are being actively worked on.

You are welcome to re-open this pull-request in the future if you do further work on this.

File: src/Search/Indexes/SearchIndex.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (michalkleiner) [reply src/Search/Indexes/SearchIndex.php:624]: This didn't help, the problem is somewhere else.

**Comment by michalkleiner** [comment]
Hi @emteknetnz, how do I reopen this PR? Looking at it code now, aiming at making the tests pass. Ed's suggestion didn't help.

**Comment by emteknetnz** [comment]
Hmm, the `Reopen and comment` button is greyed out for me, when I hover of it the tooltip says "The pulls/fix-108-getDirtyIDs branch was force-pushed or recreated"

I'd suggest either:
a) try just pushing a commit to see if it reopens, or
b) just open a new PR and link to this one

**Pull request closed without merging**